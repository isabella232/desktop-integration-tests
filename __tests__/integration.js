const Application = require("spectron").Application;
const assert = require("assert");
const path = require("path");
const fs = require("fs");

// const appPath = "nteract/applications/desktop/dist/squashfs-root/nteract";
const appPath =
  "/Users/acb/contributing/tt/nteract/applications/desktop/dist/mac/nteract.app/Contents/MacOS/nteract";
const cleanNotebookPath = "notebooks/node-example.ipynb";
const executedNotebookPath = "notebooks/node-example-executed.ipynb";
const newCodeCellNotebookPath = "notebooks/newCodeCell-example.ipynb";
const newTextCellNotebookPath = "notebooks/newTextCell-example.ipynb";
const changeThemeDarkNotebookPath = "notebooks/changeThemeDark-example.ipynb";

const delay = time => new Promise(resolve => setTimeout(resolve, time));

jest.setTimeout(15000);

let nextTestNb = executedNotebookPath;
describe("Testing notebook actions in python with dirty notebook", () => {
  beforeEach(() => {
    app = new Application({
      path: appPath,
      args: [nextTestNb]
    });
    return app.start();
  });

  afterEach(async () => {
    if (app && app.isRunning()) {
      await delay(2000);
      return app.stop();
    }
  });

  it("Open dirty notebook example, clears all cells", async () => {
    await delay(12000);
    await app.client.windowByIndex(0);
    await app.browserWindow.send("menu:clear-all");
    await app.browserWindow.send("menu:save-as", "notebooks/clear-all.ipynb");
    // Setting the path for the starting notebook for the next test. Inelegant for sure, but it works ¯\_(ツ)_/¯
    nextTestNb = cleanNotebookPath;
  });

  it("Open clean notebook example, runs all cells", async () => {
    await delay(12000);
    await app.client.windowByIndex(0);
    await app.browserWindow.send("menu:run-all");
    await app.browserWindow.send("menu:save-as", "notebooks/run-all.ipynb");
    nextTestNb = newCodeCellNotebookPath;
  });

  it("Open notebook and adds a new code cell", async () => {
    await delay(12000);
    await app.client.windowByIndex(0);
    await app.browserWindow.send("menu:new-code-cell");
    await app.browserWindow.send(
      "menu:save-as",
      "notebooks/new-code-cell.ipynb"
    );
    nextTestNb = newTextCellNotebookPath;
  });

  it("Open notebook and adds a new text cell", async () => {
    await delay(12000);
    await app.client.windowByIndex(0);
    await app.browserWindow.send("menu:new-text-cell");
    await app.browserWindow.send(
      "menu:save-as",
      "notebooks/new-text-cell.ipynb"
    );
    nextTestNb = changeThemeDarkNotebookPath;
  });

  it("Open notebook and changes the theme to dark", async () => {
    await delay(12000);
    await app.client.windowByIndex(0);
    await app.browserWindow.send("menu:theme", "dark");
    await app.browserWindow.send("menu:save-as", "notebooks/dark-theme.ipynb");
    nextTestNb = changeThemeLightNotebookPath;
  });
});

describe("tests jest snapshots of executed notebooks", () => {
  it("tests clear all-cells", () => {
    const nbPath = "notebooks/clear-all.ipynb";
    const nb = JSON.parse(fs.readFileSync(nbPath));
    expect(nb).toMatchSnapshot();
  });

  it("tests run-all-cells", () => {
    const nbPath = "notebooks/run-all.ipynb";
    const nb = JSON.parse(fs.readFileSync(nbPath));
    expect(nb).toMatchSnapshot();
  });

  it("tests new-code-cells", () => {
    const nbPath = "notebooks/new-code-cell.ipynb";
    const nb = JSON.parse(fs.readFileSync(nbPath));
    expect(nb).toMatchSnapshot();
  });

  it("tests new-text-cells", () => {
    const nbPath = "notebooks/new-text-cell.ipynb";
    const nb = JSON.parse(fs.readFileSync(nbPath));
    expect(nb).toMatchSnapshot();
  });

  it("tests change theme to dark", () => {
    const nbPath = "notebooks/dark-theme.ipynb";
    const nb = JSON.parse(fs.readFileSync(nbPath));
    expect(nb).toMatchSnapshot();
  });
});
