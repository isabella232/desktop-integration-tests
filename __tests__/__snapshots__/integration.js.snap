// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests jest snapshots of executed notebooks tests clear all-cells 1`] = `
Object {
  "cells": Array [
    Object {
      "cell_type": "markdown",
      "metadata": Object {},
      "source": Array [
        "# Dirty cells",
      ],
    },
    Object {
      "cell_type": "code",
      "execution_count": null,
      "metadata": Object {
        "collapsed": false,
        "inputHidden": false,
        "outputHidden": false,
      },
      "outputs": Array [],
      "source": Array [
        "console.log(\\"3 + 4\\")",
      ],
    },
    Object {
      "cell_type": "code",
      "execution_count": null,
      "metadata": Object {
        "collapsed": false,
        "inputHidden": false,
        "outputHidden": false,
      },
      "outputs": Array [],
      "source": Array [
        "for (var i = 0; i < 100; i++) {
",
        "    console.log(i)
",
        "}",
      ],
    },
    Object {
      "cell_type": "code",
      "execution_count": null,
      "metadata": Object {
        "collapsed": false,
        "inputHidden": false,
        "outputHidden": false,
      },
      "outputs": Array [],
      "source": Array [],
    },
  ],
  "metadata": Object {
    "kernel_info": Object {
      "name": "node_nteract",
    },
    "kernelspec": Object {
      "display_name": "Node.js (nteract)",
      "language": "javascript",
      "name": "node_nteract",
    },
    "language_info": Object {
      "file_extension": ".js",
      "mimetype": "application/javascript",
      "name": "javascript",
      "version": "8.9.3",
    },
    "nteract": Object {
      "version": "0.11.9",
    },
  },
  "nbformat": 4,
  "nbformat_minor": 4,
}
`;

exports[`tests jest snapshots of executed notebooks tests new-code-cells 1`] = `
Object {
  "cells": Array [
    Object {
      "cell_type": "markdown",
      "metadata": Object {
        "collapsed": false,
        "inputHidden": false,
        "outputHidden": false,
      },
      "source": Array [
        "## New Code Cell",
      ],
    },
    Object {
      "cell_type": "code",
      "execution_count": null,
      "metadata": Object {
        "collapsed": false,
        "inputHidden": false,
        "outputHidden": false,
      },
      "outputs": Array [],
      "source": Array [],
    },
  ],
  "metadata": Object {
    "kernel_info": Object {
      "name": "node_nteract",
    },
    "kernelspec": Object {
      "display_name": "Node.js (nteract)",
      "language": "javascript",
      "name": "node_nteract",
    },
    "language_info": Object {
      "file_extension": ".js",
      "mimetype": "application/javascript",
      "name": "javascript",
      "version": "8.9.3",
    },
    "nteract": Object {
      "version": "0.11.9",
    },
  },
  "nbformat": 4,
  "nbformat_minor": 4,
}
`;

exports[`tests jest snapshots of executed notebooks tests new-text-cells 1`] = `
Object {
  "cells": Array [
    Object {
      "cell_type": "markdown",
      "metadata": Object {
        "collapsed": false,
        "inputHidden": false,
        "outputHidden": false,
      },
      "source": Array [
        "## New Text Cell",
      ],
    },
    Object {
      "cell_type": "markdown",
      "metadata": Object {},
      "source": Array [],
    },
  ],
  "metadata": Object {
    "kernel_info": Object {
      "name": "node_nteract",
    },
    "kernelspec": Object {
      "display_name": "Node.js (nteract)",
      "language": "javascript",
      "name": "node_nteract",
    },
    "language_info": Object {
      "file_extension": ".js",
      "mimetype": "application/javascript",
      "name": "javascript",
      "version": "8.9.3",
    },
    "nteract": Object {
      "version": "0.11.9",
    },
  },
  "nbformat": 4,
  "nbformat_minor": 4,
}
`;

exports[`tests jest snapshots of executed notebooks tests run-all-cells 1`] = `
Object {
  "cells": Array [
    Object {
      "cell_type": "markdown",
      "metadata": Object {},
      "source": Array [
        "# Clean cells",
      ],
    },
    Object {
      "cell_type": "code",
      "execution_count": 1,
      "metadata": Object {
        "collapsed": false,
        "inputHidden": false,
        "outputHidden": false,
      },
      "outputs": Array [
        Object {
          "name": "stdout",
          "output_type": "stream",
          "text": Array [
            "3 + 4
",
          ],
        },
      ],
      "source": Array [
        "console.log(\\"3 + 4\\")",
      ],
    },
    Object {
      "cell_type": "code",
      "execution_count": 2,
      "metadata": Object {
        "collapsed": false,
        "inputHidden": false,
        "outputHidden": false,
      },
      "outputs": Array [
        Object {
          "name": "stdout",
          "output_type": "stream",
          "text": Array [
            "0
",
            "1
",
            "2
",
            "3
",
            "4
",
            "5
",
            "6
",
            "7
",
            "8
",
            "9
",
            "10
",
            "11
",
            "12
",
            "13
",
            "14
",
            "15
",
            "16
",
            "17
",
            "18
",
            "19
",
            "20
",
            "21
",
            "22
",
            "23
",
            "24
",
            "25
",
            "26
",
            "27
",
            "28
",
            "29
",
            "30
",
            "31
",
            "32
",
            "33
",
            "34
",
            "35
",
            "36
",
            "37
",
            "38
",
            "39
",
            "40
",
            "41
",
            "42
",
            "43
",
            "44
",
            "45
",
            "46
",
            "47
",
            "48
",
            "49
",
            "50
",
            "51
",
            "52
",
            "53
",
            "54
",
            "55
",
            "56
",
            "57
",
            "58
",
            "59
",
            "60
",
            "61
",
            "62
",
            "63
",
            "64
",
            "65
",
            "66
",
            "67
",
            "68
",
            "69
",
            "70
",
            "71
",
            "72
",
            "73
",
            "74
",
            "75
",
            "76
",
            "77
",
            "78
",
            "79
",
            "80
",
            "81
",
            "82
",
            "83
",
            "84
",
            "85
",
            "86
",
            "87
",
            "88
",
            "89
",
            "90
",
            "91
",
            "92
",
            "93
",
            "94
",
            "95
",
            "96
",
            "97
",
            "98
",
            "99
",
          ],
        },
      ],
      "source": Array [
        "for (var i = 0; i < 100; i++) {
",
        "    console.log(i)
",
        "}",
      ],
    },
    Object {
      "cell_type": "code",
      "execution_count": 3,
      "metadata": Object {
        "collapsed": false,
        "inputHidden": false,
        "outputHidden": false,
      },
      "outputs": Array [],
      "source": Array [],
    },
  ],
  "metadata": Object {
    "kernel_info": Object {
      "name": "node_nteract",
    },
    "kernelspec": Object {
      "display_name": "Node.js (nteract)",
      "language": "javascript",
      "name": "node_nteract",
    },
    "language_info": Object {
      "file_extension": ".js",
      "mimetype": "application/javascript",
      "name": "javascript",
      "version": "8.9.3",
    },
    "nteract": Object {
      "version": "0.11.9",
    },
  },
  "nbformat": 4,
  "nbformat_minor": 4,
}
`;
